aws/api-servers.tf

```diff
@@ -3,17 +3,19 @@ resource "aws_launch_configuration" "api" {
   image_id = "${lookup(var.amis, var.region)}"
   instance_type = "t2.medium"
   security_groups = ["${aws_security_group.default.id}"]
-  key_name = "${aws_key_pair.deployer.key_name}"
+  key_name = "${var.key_pair_name}"
+/*
```

Comment:
I'm not sure, it's a dual problem...
Without the lifecycle option, we cannot *update* launch configurations after they've been created (i.e.: change AMI, instance size, etc).
But with them, we cannot *create* new environments because of the cycle dependency bug...
Maybe adding a comment explaining why is't commented?